File: Easy/57- 268. Missing Number/Arithmetic progression.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
    let sum = (nums.length * (nums.length + 1)) / 2;
    nums.forEach((element) => {
        sum -= element;
    });
    return sum;
};

File: Easy/57- 268. Missing Number/Arithmetic progression.cs
public class Solution
{
    public int MissingNumber(int[] nums)
    {
        // O(1) maths law to get sum of values from 0 to n
        int sum = nums.Length * (nums.Length + 1) / 2;

        for (int i = 0; i < nums.Length; i++)
        {
            // subtract a value at each iteration
            sum -= nums[i];
        }

        // we end up with only the missing value
        return sum;
    }
}